src/node_task_runner.cc

```diff
@@ -0,0 +1,219 @@
+#include "node_task_runner.h"
+#include "util.h"
+
+#include <regex>  // NOLINT(build/c++11)
+
+namespace node::task_runner {
+
+#if defined(_WIN32)
+static const char separator[] = "\\";
+static const char bin_path[] = "node_modules\\.bin";
+static const char task_file[] = "cmd.exe";
+#else
+static const char separator[] = "/";
+static const char bin_path[] = "node_modules/.bin";
+static const char task_file[] = "/bin/sh";
+#endif  // defined(_WIN32)
+
+ProcessRunner::ProcessRunner(
+    std::unique_ptr<InitializationResultImpl>* result,
+    std::string_view command,
+    const std::optional<std::string>& positional_args) {
+  process_ = {};
+  loop = uv_default_loop();
+  memset(&options, 0, sizeof(uv_process_options_t));
+
+  // Inherit stdin, stdout, and stderr from the parent process.
+  options.stdio_count = 3;
+  child_stdio[0].flags = UV_INHERIT_FD;
+  child_stdio[0].data.fd = 0;
+  child_stdio[1].flags = UV_INHERIT_FD;
+  child_stdio[1].data.fd = 1;
+  child_stdio[2].flags = UV_INHERIT_FD;
+  child_stdio[2].data.fd = 2;
+  options.stdio = child_stdio;
+  options.exit_cb = ExitCallback;
+  options.file = task_file;
+
+  init_result = result;
+  process_.data = this;
+
+  // Get the current working directory.
+  char cwd[PATH_MAX_BYTES];
+  size_t cwd_size = PATH_MAX_BYTES;
+  CHECK_EQ(uv_cwd(cwd, &cwd_size), 0);
+  CHECK_GT(cwd_size, 0);
+
+  // Example: "/Users/user/project/node_modules/.bin"
+  std::string current_bin_path = cwd + std::string(separator) + bin_path;
+
+  std::string command_str = std::string(command);
```

Comment:
```suggestion
  std::string command_str(command);
```
Is there a reason to not do this?